controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java

```diff
@@ -405,17 +409,42 @@ private boolean servicesConverged(ApplicationId id, JobType type, DualLogger log
                 return Optional.empty();
             case FAILURE:
                 logger.log("Tests failed.");
-                status = testFailure; break;
+                return Optional.of(testFailure);
             case ERROR:
                 logger.log(INFO, "Tester failed running its tests!");
-                status = error; break;
+                return Optional.of(error);
             case SUCCESS:
                 logger.log("Tests completed successfully.");
-                status = running; break;
+                return Optional.of(running);
             default:
                 throw new IllegalStateException("Unknown status '" + testStatus + "'!");
         }
-        return Optional.of(status);
+    }
+
+    private Optional<RunStatus> copyVespaLogs(RunId id, DualLogger logger) {
+        ZoneId zone = id.type().zone(controller.system());
+        logger.log("Copying Vespa log from nodes of " + id.application() + " in " + zone + " ...");
+        try {
+            List<LogEntry> entries = new ArrayList<>();
+            String logs = IOUtils.readAll(controller.configServer().getLogStream(new DeploymentId(id.application(), zone),
+                                                                                 Collections.emptyMap()), // Get all logs.
+                                          StandardCharsets.UTF_8);
+            for (String line : logs.split("\n")) {
+                String[] parts = line.split("\t");
+                if (parts.length != 7) continue;
+                entries.add(new LogEntry(0,
+                                         (long) (Double.parseDouble(parts[0]) * 1000),
+                                         LogEntry.typeOf(LogLevel.parse(parts[5])),
+                                         parts[1] + '\t' + parts[3] + '\t' + parts[4] + '\n' +
+                                         parts[6].replaceAll("\\\\n", "\n")
+                                                 .replaceAll("\\\\t", "\t")));
```

Comment:
Shouldn't this be `parts[6].replaceAll("\\n", "\n").replaceAll("\\t", "\t")));`?

Reply:
No, it's a regex, so needs double quoting :)